test/cpp/c10d/CMakeLists.txt

```diff
@@ -10,9 +10,6 @@ function(c10d_add_test test_src)
   add_executable(${test_name} "${test_src}")
   target_include_directories(${test_name} PRIVATE $<BUILD_INTERFACE:${TORCH_SRC_DIR}/csrc/distributed>)
   target_link_libraries(${test_name} ${ARGN})
-  if(NOT WIN32)
-    target_link_libraries(${test_name} pthread)
-  endif()
```

Comment:
I don't think CI tests that PyTorch can be build with Distributed for windows, so please provide some sort of test plan

Reply:
Because pthread is already linked by other targets. However, I will revert it to simplify this PR.